rclcpp/test/rclcpp/node_interfaces/test_node_graph.cpp

```diff
@@ -66,18 +66,57 @@ class TestNodeGraph : public ::testing::Test
 
   const rclcpp::node_interfaces::NodeGraph * node_graph() const {return node_graph_;}
 
+  size_t get_num_topics()
+  {
+    constexpr std::chrono::milliseconds timeout(100);
+
+    int tries = 0;
+    size_t num_topics = 0;
+    while (num_topics < 1u && tries++ < 5) {
+      auto event = node()->get_graph_event();
+      EXPECT_NO_THROW(node()->wait_for_graph_change(event, timeout));
+
+      auto topic_names_and_types = node_graph()->get_topic_names_and_types();
+      num_topics = topic_names_and_types.size();
```

Comment:
@clalancette meta: having some form of `wait_for_graph()` function taking the node graph, a predicate on that node graph, and a timeout would be neat.

Reply:
Yeah, true.  I did consider it, but I wasn't sure there were enough uses in here to warrant it.  But if you think it is clearer, I can do that.  Thoughts?